includes/class-sensei-dependency-checker.php

```diff
@@ -17,33 +17,71 @@
  * @since 2.0.0
  */
 class Sensei_Dependency_Checker {
-	const MINIMUM_PHP_VERSION = '7.0';
+	const MINIMUM_PHP_VERSION        = '7.0';
+	const FUTURE_MINIMUM_PHP_VERSION = '7.2';
 
 	/**
 	 * Checks for our PHP version requirement.
 	 *
 	 * @return bool
 	 */
-	public static function check_php() {
-		return version_compare( phpversion(), self::MINIMUM_PHP_VERSION, '>=' );
+	public static function check_php_requirement() {
+		return self::verify_php( self::MINIMUM_PHP_VERSION );
 	}
 
 	/**
-	 * Adds notice in WP Admin that minimum version of PHP is not met.
+	 * Checks for our future PHP version requirement.
+	 *
+	 * @return bool
+	 */
+	public static function check_future_php_requirement() {
+		return self::verify_php( self::FUTURE_MINIMUM_PHP_VERSION );
+	}
+
+	/**
+	 * Checks for our PHP version requirement.
+	 *
+	 * @return bool
+	 */
+	private static function verify_php( $version ) {
+		return version_compare( phpversion(), $version, '>=' );
+	}
+
+	/**
+	 * Adds error in WP Admin that minimum version of PHP is not met.
+	 *
+	 * @access private
+	 */
+	public static function add_php_error() {
+		// translators: %1$s is version of PHP that Sensei requires; %2$s is the version of PHP WordPress is running on.
+		$message = sprintf( __( '<strong>Sensei LMS</strong> requires a minimum PHP version of %1$s, but you are running %2$s.', 'sensei-lms' ), self::MINIMUM_PHP_VERSION, phpversion() );
+		self::show_php_notice( $message );
+	}
+
+	/**
+	 * Adds warning in WP Admin that the future minimum version of PHP is not met.
 	 *
 	 * @access private
 	 */
 	public static function add_php_notice() {
+		// translators: %1$s is version of PHP that Sensei is going to require in the future; %2$s is the version of PHP WordPress is running on.
+		$message = sprintf( __( '<strong>Sensei LMS</strong> will require, in the next release, a minimum PHP version of %1$s, but you are running %2$s.', 'sensei-lms' ), self::FUTURE_MINIMUM_PHP_VERSION, phpversion() );
```

Comment:
Can we do anything to make "the next release" something more concise?
Also, maybe it would be nice to provide a link on [how to update PHP](https://wordpress.org/support/update-php/).

Reply:
Maybe we can point to [here](https://senseilms.com/documentation/faq/#requirements) after updating that FAQ :)

Reply:
One thing I forgot to ask about yesterday, and I'm not sure if THAT'S important: Do you think this notice should be dismissable? I mean, right now, it will appear in the Notices section on wp-admin and it can't be hidden...so I think it's worth asking.

WDYT @renatho @aaronfc?